packages/css/src/transformCss.ts

```diff
@@ -508,6 +556,11 @@ class Stylesheet {
   toCss() {
     const css: Array<string> = [];
 
+    // Render layer declarations first
+    for (const layerDeclaration of this.layerDeclarations) {
+      css.push(renderCss({ [`@layer ${layerDeclaration.name}`]: EMPTY }));
```

Comment:
This implementation forces all layers to be hoisted. I'm not sure that's what we want. I know we play with font faces and media query order but I think those cases are different and we should respect the order the user specified?